roles/tobiko/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+  ansible.builtin.import_role:
+    name: tempest
+    tasks_from: create-clouds-file
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Copy CA bundle to cifmw_tobiko_artifacts_basedir
+  ansible.builtin.copy:
+    src: "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+    dest: "{{ cifmw_tobiko_artifacts_basedir }}"
+    mode: '0444'
+    owner: "{{ lookup('env', 'USER') }}"
+    group: "{{ lookup('env', 'USER') }}"
+    remote_src: true
+
+- name: Set proper permission for tobiko directory
+  ansible.builtin.command:
+    cmd: "podman unshare chown 42495:42495 -R {{ cifmw_tobiko_artifacts_basedir }}"
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Ensure we have tobiko container image
+  register: _tobiko_fetch_img
+  containers.podman.podman_image:
+    name: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+  retries: 5
+  delay: 5
+  until: _tobiko_fetch_img is success
+
+- name: Run tobiko
+  ignore_errors: true
+  containers.podman.podman_container:
+    name: tobiko
+    image: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+    state: started
+    auto_remove: "{{ cifmw_tobiko_remove_container }}"
+    network: host
+    volume:
+      - "{{ cifmw_tobiko_artifacts_basedir }}/:/var/lib/tobiko/external_files:Z"
+      - "{{ cifmw_tobiko_artifacts_basedir }}/tls-ca-bundle.pem:/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem:Z"
+    detach: false
+    dns: "{{ cifmw_tobiko_dns_servers }}"
+    env:
+      TOBIKO_DEBUG: "{{ ci_tobiko_debug }}"
+      TOBIKO_TESTENV: "{{ ci_tobiko_testenv }}"
+      TOBIKO_PREVENT_CREATE: "{{ ci_tobiko_prevent_create if ci_tobiko_prevent_create is not none else omit }}"
+      TOBIKO_VERSION: "{{ ci_tobiko_version if ci_tobiko_version is not none else omit }}"
+      TOBIKO_UBUNTU_INTERFACE_NAME: "{{ ci_tobiko_ubuntu_interface_name if ci_tobiko_ubuntu_interface_name is not none else omit }}"
+      TOBIKO_KEYSTONE_INTERFACE: "{{ ci_tobiko_keystone_interface if ci_tobiko_keystone_interface is not none else omit }}"
+      TOBIKO_TESTCASE_TIMEOUT: "{{ ci_tobiko_testcase_timeout if ci_tobiko_testcase_timeout is not none else omit }}"
+      TOBIKO_TESTRUNNER_TIMEOUT: "{{ ci_tobiko_testrunner_timeout if ci_tobiko_testrunner_timeout is not none else omit }}"
+      TOBIKO_PYTEST_ADDOPTS: "{{ ci_tobiko_pytest_addopts if ci_tobiko_pytest_addopts is not none else omit }}"
+      TOBIKO_REPORT_DIR: "{{ ci_tobiko_report_dir if ci_tobiko_report_dir is not none else omit }}"
+      TOBIKO_MANILA_SHARE_PROTOCOL: "{{ ci_tobiko_manila_share_protocol if ci_tobiko_manila_share_protocol is not none else omit }}"
+  when: not cifmw_tobiko_dry_run | bool
+  register: tobiko_run_output
+
+- name: Change tobiko directory permission back to original
+  become: true
+  ansible.builtin.file:
+    path: "{{ cifmw_tobiko_artifacts_basedir }}"
+    state: directory
+    recurse: true
+    owner: "{{ lookup('env', 'USER') }}"
+    group: "{{ lookup('env', 'USER') }}"
+
+- name: Save logs from podman
```

Comment:
wondering if we couldn't use the internal podman logging facility to avoid loading the ansible process memory.
For instance, using [log_driver](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-log_driver) (and setting it to k8s iirc), in addition to [log_opt](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-log_opt) (and setting it to `path: "{{ cifmw_tobiko_artifacts_basedir }}/podman_tobiko.log"` for instance) may be better/faster.
We can also play with the [log_level](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-log_level) while at it.

Might be a follow-up, iirc Tempest role is doing the same.

Reply:
My impression is that saving the podman logs was implemented this way on the tempest role too because of problems with the permissions.
The `cifmw_tobiko_artifacts_basedir` owner is changed in line 47.
Original ownership is restored in line 92.

If the task `Run tobiko` tries to create a file under that directory, it will fail.

Reply:
possible. I'll try to get some time to some tests. Maybe letting podman write to its standard log location and copying it is better - as said, potential follow-up.